firrtl/src/main/scala/firrtl/ir/Serializer.scala

```diff
@@ -96,7 +96,9 @@ object Serializer {
     case ValidIf(cond, value, _) => b ++= "validif("; s(cond); b ++= ", "; s(value); b += ')'
     case SIntLiteral(value, width) =>
       b ++= "SInt"; s(width); b ++= "(\"h"; b ++= value.toString(16); b ++= "\")"
-    case other => b ++= other.serialize // Handle user-defined nodes
+    case ProbeExpr(expr, _) => b ++= "probe("; s(expr); b += ')'
```

Comment:
If ProbeExpr is writable, this should be `rwprobe` ?

Reply:
totally right. I think I actually need to create an entirely different expression here.